main.py

```diff
@@ -0,0 +1,267 @@
+		self.width = width
+		self.height = height
+		self.dx = dx
+		self.dy = dy
+		#shiftdxs are for moving everything when the player moves
+		self.shiftdx = shiftdx
+		self.jumpdy = jumpdy # variable dy is set to when controller jumps
+
+	def train_wreck(self, train):
+		#train collision function
+		return (train.x+train.width) > self.x
+
+	def shift_world(self):
+		#shift world to left when past a certain point
+		return self.x > 350
+
+	def go_back(self):
+		#shift world to right when past a certain point
+		return self.x < 130
+
+	def hit_ground(self,ground):
+		#ground collision function for jumping
+		return (self.y + self.height) > ground.y and self.x < (ground.x+ground.width) and self.x > ground.x
+
+	def off_screen(self):
+		#if player goes off screen
+		return self.y > 480
+
+class PainTrain(object):
+	def __init__(self,x=0,y=0,width=200,height=200,constdx=.05,dx=0,shiftdx=-1):
+		# places train centered above coordinate given
+		self.x = x
+		self.y = y-height
+		#size of chasing cube
+		self.width = width
+		self.height = height
+		#constant velocity
+		self.constdx = constdx
+		self.dx = dx
+		self.shiftdx = shiftdx
+
+	def step(self):
+		self.x += self.constdx
+
+# classes for level objects
+class Ground(object):
+	def __init__(self, x = 0, y = 300, width = 2400, height = 180,dx=0,shiftdx=-1):
+		self.x = x
+		self.y = y
+		#size of ground
+		self.width = width
+		self.height = height
+		self.dx = dx
+		#shiftdx is for moving the ground when player moves
+		self.shiftdx = shiftdx
+
+class Platform(object):
+	def __init__(self, x=0,y=0,width = 100, height = 20, dx=0, shiftdx=-1):
+		self.x = x
+		self.y = y
+		self.width = width
+		self.height = height
+		self.dx = dx
+		#shiftdx is for moving the ground when player moves
+		self.shiftdx = shiftdx
+
+"""View classes"""
+class PlayerView(object):
+	def __init__(self, model):
+		self.model = model
+
+	def draw(self, surface):
+		model = self.model
+		# this takes (x,y,width,height)
+		pygame.draw.rect(surface,SLATEGRAY,(model.x,model.y,model.width,model.height))
+
+class PainTrainView(object):
+	def __init__(self, model):
+		self.model = model
+
+	def draw(self, surface):
+		model = self.model
+		# this takes (x,y,width,height)
+		pygame.draw.rect(surface,BLACK,(model.x,model.y,model.width,model.height))
+
+class GroundView(object):
+	def __init__(self, model):
+		self.model = model
+
+	def draw(self, surface):
+		model = self.model
+		# this takes (x,y,width,height)
+		pygame.draw.rect(surface,DIMGRAY,(model.x,model.y,model.width,model.height))
+
+class ObstacleView(object):
+	# can be used for any rectangular object
+	def __init__(self,model):
+		self.model = model
+
+	def draw(self,surface):
+		model = self.model
+		pygame.draw.rect(surface,BLACK,(model.x,model.y,model.width,model.height))
+
+"""Controller classes"""
+class Controller(object):
+	def __init__(self,models):
+		self.models = models
+		self.player = models[0] # make sure this aligns with controlled_models in main
```

Comment:
A good idea for the future: pass in the player separately from the list of models.

This is fine, but put a docstring saying that the player must be the first model in the list (along with other, typical docstring stuff)